signer/src/codec.rs

```diff
@@ -76,32 +89,79 @@ pub trait Decode: Sized {
     fn decode<R: io::Read>(reader: R) -> Result<Self, Error>;
 }
 
-impl<T: serde::Serialize> Encode for &T {
-    fn encode<W: io::Write>(self, writer: W) -> Result<(), Error> {
-        bincode::serialize_into(writer, self)
+impl<T> Encode for T
+where
+    T: ProtoSerializable + Clone,
+    T: Into<<T as ProtoSerializable>::Message>,
+{
+    fn encode<W: io::Write>(self, _writer: W) -> Result<(), Error> {
+        unimplemented!()
```

Comment:
We actually use `encode_to_vec` around. I'll change the default impl of `encode` to use `encode_to_vec`, since with protobuf we can have an infallible vec encoding (the thing Daniel mentioned before) 🚀

Reply:
Removed the unused method and refactored things in [4bad83b](https://github.com/stacks-network/sbtc/pull/880/commits/4bad83b420d85baf6721335d4a5cc7843187de4b)